app/configurator/configurator-state/actions.tsx

```diff
@@ -91,7 +91,16 @@ export type ConfiguratorStateAction =
     }
   | {
       type: "CHART_PALETTE_CHANGED";
-      value: NewColorField;
+      value: {
+        field: string;
+        colorConfigPath?: string;
+        palette: string;
+        colorMapping: Record<string, string>;
+      };
+    }
+  | {
+      type: "CHART_PALETTE_CHANGED_NEW";
+      value: ColorField;
     }
```

Comment:
@noahonyejese I wonder if we can have a bit better separation here, I think like the `CHART_PALETTE_CHANGED_NEW` action is actually `COLOR_FIELD_SET` and `CHART_PALETTE_CHANGED` is more `COLOR_MAPPING_UPDATED`?

Maybe the `colorConfigPath` could be required and this action only used for maps and tables, the rest of the charts using `COLOR_FIELD_SET`? Also, I think `palette` should be `paletteId`?

Not sure if this is the best idea, let me know what you think :)

Reply:
@bprusinowski You mean adding the `colorConfigPath` to COLOR_FIELD_SET?

Reply:
I thought about either having one action that would have `colorConfigPath` (I think as we had before), or to make `colorConfigPath` non-optional – it's it the case that we always need to set it? If no, then might be okay as it is, I just remember some bugs from the past related to this property, so making it a bit more bullet-proof would be fantastic 🥹